docs/rpc@usage/mainserver.py

```diff
@@ -0,0 +1,64 @@
+# saved as greeting-server.py
+import Pyro4
+import subprocess #Needed to run a shell command or script from python
+import shlex #Used when a shell command is to be splitted into sections. Like below.
+import json
+import unittest
+
+class MainServer(object):
+    def __init__(self):
+	#Loads the variable dict with the contents of config.json.
+	self.dict = json.loads(open('config.json').read())
+	self.server_details = self.dict['server-details'] #The server ip addresses.
+	self.script_list = self.dict['script-list'] #The script name and no. of arguments.
+	#This is the ip address of the machine on which the host is binded on.
+	Pyro4.config.HOST = self.server_details['server-address']
+
+    #Takes in the list_of_commands passed to the the Server program and concatentes everything 
+    #into a single string
+    def concatenate_command(self, list_of_commands):
+	concatenated_command = self.script_list[list_of_commands[0]]['name'] + " "
+        concatenated_command += " ".join(list_of_commands[1:])
+	return concatenated_command
+    
+    #Splits the concatenated_command and returns a list of all the space separated commands in the
+    def split_command(self, concatenated_command):
+	return shlex.split(concatenated_command)
+    
+    #Takes in the concatenated_command, splits it and checks whether the number of argument
+    #is same as that of required by the program.
+    def correct_argument_list_length(self, concatenated_command, file_key_in_dict):
+	list_m = self.split_command(concatenated_command)
+        return (len(list_m)-1 == int(self.script_list[file_key_in_dict]['arguments']))
+    
+    #This method checks for escape characters or injection attacks.
+    def escape_characters_present(self, concatenated_command):
+	#The escape_chars set contains all the characters which if present would terminate the program
+	#execution	
+	escape_chars = set(';<>&:|\'')
+	if any((c in escape_chars) for c in concatenated_command):
+		return True
+	else:
+		return False
+
+    #Takes in the list_of_commands passed by the client program.
+    def run_script(self, list_of_commands):
```

Comment:
Yes, You're right. For now since we have to work with shellscripts, we can use this. Later when we have something better in python, we can invoke the remotely.